packages/easy-coding-standard/tests/Error/ErrorCollector/SniffFileProcessorReportWarningTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\EasyCodingStandard\Tests\Error\ErrorCollector;
+
+use PHP_CodeSniffer\Standards\Generic\Sniffs\CodeAnalysis\ForLoopShouldBeWhileLoopSniff;
+use Symplify\EasyCodingStandard\Caching\ChangedFilesDetector;
+use Symplify\EasyCodingStandard\HttpKernel\EasyCodingStandardKernel;
+use Symplify\EasyCodingStandard\SniffRunner\Application\SniffFileProcessor;
+use Symplify\EasyCodingStandard\SniffRunner\ValueObject\Error\CodingStandardError;
+use Symplify\EasyCodingStandard\ValueObject\Configuration;
+use Symplify\EasyCodingStandard\ValueObject\Error\FileDiff;
+use Symplify\PackageBuilder\Testing\AbstractKernelTestCase;
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+final class SniffFileProcessorReportWarningTest extends AbstractKernelTestCase
+{
+    private SniffFileProcessor $sniffFileProcessor;
+
+    protected function setUp(): void
+    {
+        $this->bootKernelWithConfigs(
+            EasyCodingStandardKernel::class,
+            [__DIR__ . '/SniffRunnerSource/config-report-warnings.php']
+        );
+
+        $this->sniffFileProcessor = $this->getService(SniffFileProcessor::class);
+
+        $changedFilesDetector = $this->getService(ChangedFilesDetector::class);
+        $changedFilesDetector->clearCache();
+    }
+
+    /** @return mixed[][] */
+    public function provider(): array
+    {
+        return [
+            'no report' => [0, new Configuration()],
+            'report is set up' => [1, new Configuration(reportWarnings: [ForLoopShouldBeWhileLoopSniff::class])],
```

Comment:
The `ForLoopShouldBeWhileLoopSniff` should be added in https://github.com/symplify/symplify/pull/3551/files#diff-de10631f76c115195abacad48f6d75fbe7bfaf7acce0e92cfc9cd51bf1f2cfd6R32 `$reportWarnings`.

Reply:
Ok, I will append it and choose other sniff for testing.

Reply:
I've manage to create a FakeSniff

Reply:
Resolved on commits 52c36f3 and 686ece2